docs/operation/troubleshooting.md

```diff
@@ -0,0 +1,79 @@
+---
```

Comment:
This is missing a _lot_ of stuff including:

* Requesting standalone replication when contacting support
* Reliable replication and its importance to helping troubleshoot
* Details on why deadlock occurs
* References to how debugging can help troubleshoot (including overriding deadlock detection)
* References to how logging/metrics can help troubleshoot 
* References to how unit tests and real-server tests can help troubleshoot
* References to how static analysis can help troubleshoot